Spanner/tests/Perf/Operation.php

```diff
@@ -0,0 +1,267 @@
+<?php
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Spanner\Tests\Perf;
+
+use Google\Cloud\Spanner\Database;
+
+class Operation
+{
+    /**
+     * @var string
+     */
+    private $database;
+
+    /**
+     * @var array
+     */
+    private $parameters;
+
+    /**
+     * float
+     */
+    private $totalWeight;
+
+    /**
+     * @var array
+     */
+    private $weights;
+
+    /**
+     * @var array
+     */
+    private $operations;
+
+    /**
+     * @var array
+     */
+    private $latency;
+
+    /**
+     * @var array
+     */
+    private $keys;
+
+    /**
+     * @param string $database
+     * @param array $parameters
+     * @param float $totalWeight
+     * @param array $weights
+     * @param array $operations
+     * @param array $latency
+     */
+    public function __construct(
+        $database,
+        array $parameters,
+        $totalWeight,
+        array $weights,
+        array $operations,
+        array $latency
+    ) {
+        $this->database = $database;
+        $this->parameters = $parameters;
+        $this->totalWeight = $totalWeight;
+        $this->weights = $weights;
+        $this->operations = $operations;
+        $this->latency = $latency;
+    }
+
+    public function load()
+    {
+        $this->keys = [];
+
+        $startTime = microtime(true);
+        $snapshot = $this->database->snapshot();
+
+        // Kind of assuming that id is always name of PK in whatever table you choose
+        $results = $snapshot->execute('SELECT id FROM ' . $this->parameters['table']);
+        foreach ($results as $row) {
+            $this->keys[] = $row['id'];
+        }
+
+        return microtime(true) - $startTime;
+    }
+
+    /**
+     * Run a single thread of the workload
+     *
+     * @return void
+     */
+    public function run()
+    {
+        if (!is_array($this->keys)) {
+            throw new \RuntimeException('Call Operation::load() first.');
```

Comment:
Is there a need to decouple `load`/`run`? If they're reliant on each other, could we just make `load` private and call it from `run`?

Reply:
Only because the initial implementation reported the load time separately from the run (i.e. not in the `latency` array). I kept this paradigm, not knowing whether it was useful or not.

Reply:
I didn't mean to resolve this. Looking at the initial implementation the line to output the load time is commented out. I don't see any harm in sharing that information, so how about we leave in the calculation for load time and actually report it?